src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -2151,6 +2142,73 @@ JNI_METHOD(jobject, computePaseVerifier)
     return nullptr;
 }
 
+JNI_METHOD(jobject, getICDClientInfo)(JNIEnv * env, jobject self, jlong handle, jint jFabricIndex)
+{
+    chip::DeviceLayer::StackLock lock;
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    jobject jInfo = nullptr;
+    chip::app::ICDClientInfo info;
+    chip::FabricIndex fabricIndex = static_cast<chip::FabricIndex>(jFabricIndex);
+
+    ChipLogProgress(Controller, "getICDClientInfo(%u) called", fabricIndex);
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+    VerifyOrReturnValue(wrapper != nullptr, nullptr, ChipLogError(Controller, "wrapper is null"));
+
+    err = JniReferences::GetInstance().CreateArrayList(jInfo);
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, nullptr,
+                        ChipLogError(Controller, "CreateArrayList failed!: %" CHIP_ERROR_FORMAT, err.Format()));
+
+    auto iter = wrapper->getICDClientStorage()->IterateICDClientInfo();
+
+    jmethodID constructor;
+    jclass infoClass;
+
+    err = JniReferences::GetInstance().GetClassRef(env, "chip/devicecontroller/ICDClientInfo", infoClass);
```

Comment:
please kindly stop using GetClassRef and s JniClass icdClientInfoClass pattern.
please use GetLocalClassRef.
Although this  getICDClientInfo is called from java/kotlin layer and JVM do manage the local reference recycling, I prefer we explicitly call  JniLocalReferenceManager manager(env) in this beginning of the function.

Reply:
I modified.

Reply:
thanks